lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -626,6 +626,20 @@ void nrf_modem_os_init(void)
 #endif
 }
 
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
+/* The value of DELAY_LOOP_VAL is found experimentally to introduce the right amount of delay
+ * for the RTT client on host side to fetch the modem traces in time and thereby reduce data loss.
+ * The delay introduced by this function is about 4.7 ms.
+ */
+#define DELAY_LOOP_VAL 75000
+static void rtt_read_delay(void)
+{
+	for (int i = 0; i < DELAY_LOOP_VAL; i++) {
```

Comment:
I'd rather use `k_busy_wait()` instead of writing the function on your own. 

Another thing is, does the delay has to be a busy loop? 4.7 ms sounds quite long, perhaps we could use `k_sleep()` instead and let other threads to run in a meantime?

Reply:
This piece of code runs in interrupt context. So we cant use `k_sleep()`. But soon we will do trace processing in a thread and that would help us in these situations

Reply:
> But soon we will do trace processing in a thread and that would help us in these situations

That would be nice, because blocking for ~5 ms in an interrupt doesn't sound good either.